### PR TITLE
cses 2138 uwu

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cses-2138.mdx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+const int MAXN = 5e4;
+
+vector<int> graph[MAXN + 1];
+bitset<MAXN + 1> can[MAXN + 1];
+vector<bool> visited(MAXN + 1);
```
Comment (SansPapyrus683) [inline solutions/cses-2138.mdx:35]: just make everything 0-indexed
how hard can it be

File: solutions/cses-2138.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+void dfs(int node) {
```
Comment (SansPapyrus683) [inline solutions/cses-2138.mdx:37]: well yes, this is a dfs
what does it do tho

File: solutions/cses-2138.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	visited[node] = true;
+
+	// base case: node can reach itself
+	can[node][node] = 1;
+
+	for (int i: graph[node]) {
```
Comment (SansPapyrus683) [inline solutions/cses-2138.mdx:43]:
```suggestion
	for (int i : graph[node]) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
your mohter

File: solutions/cses-2138.mdx (hunk 2 above)
Comment (envyaims) [reply solutions/cses-2138.mdx:37]: its explained in the explanation and there are comments. what more explanation do you need?

File: solutions/cses-2138.mdx (hunk 1 above)
Comment (envyaims) [reply solutions/cses-2138.mdx:35]: does it matter

i just used 1-indexed because the statement is 1-indexed

File: solutions/cses-2138.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/cses-2138.mdx:37]: and yet you don’t like it when variable names are bad even when they have comments
ok then

**Review by SansPapyrus683 — APPROVED** [review]
i give up

**Comment by SansPapyrus683** [comment]
will merge once another person approves

**Review by jjliewie — APPROVED** [review]

**Comment by jjliewie** [comment]
looks good to me

**Pull request merged**